git_machete/git_operations.py

```diff
@@ -754,11 +754,15 @@ def merge_fast_forward_only(self, branch: LocalBranchShortName) -> None:  # refs
 
     def rebase(self, onto: AnyRevision, fork_revision: AnyRevision, branch: LocalBranchShortName, opt_no_interactive_rebase: bool) -> None:
         def do_rebase() -> None:
+            # Let's use `OPTS` suffix for consistency with git's built-in env var `GIT_DIFF_OPTS`
+            git_machete_rebase_opts_var = 'GIT_MACHETE_REBASE_OPTS'
+            rebase_opts_str = os.environ.get(git_machete_rebase_opts_var)
+            rebase_opts = filter(None, rebase_opts_str.split(' ')) if rebase_opts_str else []
```

Comment:
May I propose:

```suggestion
            rebase_opts_str = os.environ.get(git_machete_rebase_opts_var, "")
            rebase_opts = [option for option in rebase_opts_str.split() if option]
```

- `.get` can return default. Another benefit, if #118 is pursued, is that `rebase_opts_str: str` instead of `rebase_opts_str: Optional[str]`
  > Return the value for key if key is in the dictionary, else default. If default is not given, it defaults to None, so that this method never raises a KeyError.

- I'm not sure if it's a common knowledge what `filter(None, …)` does
  > Note that filter(function, iterable) is equivalent to the generator expression (item for item in iterable if function(item)) if function is not None and (item for item in iterable if item) if function is None.

- `split()` is probably better:
  > If sep is not specified or is None, a different splitting algorithm is applied: runs of consecutive whitespace are regarded as a single separator, and the result will contain no empty strings at the start or end if the string has leading or trailing whitespace. Consequently, splitting an empty string or a string consisting of just whitespace with a None separator returns [].

Reply:
Good spots, all three